### PR TITLE
Refactor probers part

### PR DESCRIPTION
**Pull request by rstm-sf**
refactoring to understand how to submit probers

### CODE REVIEW

**Comment by 304NotModified** [comment]
Hi

Sorry for the delay. Will try to look at this soon. Feel free to remind me later.

File: src/DetectionResult.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public DetectionResult(DetectionDetail detectionDetail)
         /// <summary>
         /// All results
         /// </summary>
-        public IList<DetectionDetail> Details { set; get; }
+        public IList<DetectionDetail> Details { get; private set; }
```
Comment (304NotModified) [inline src/DetectionResult.cs]: Good idea, but this is a breaking change. So please leave this now as it is.

Comment (rstm-sf) [reply src/DetectionResult.cs]: Ok, np :)

**Comment by 304NotModified** [comment]
Thanks!

**Pull request merged**